src/utransport.rs

```diff
@@ -63,158 +36,35 @@ fn spawn_nonblock_callback(listener: &Arc<dyn UListener>, listener_msg: UMessage
-                    };
-                    let u_attribute = match UPTransportZenoh::attachment_to_uattributes(attachment)
-                    {
-                        Ok(uattr) => uattr,
-                        Err(e) => {
-                            warn!("Unable to transform attachment to UAttributes: {e:?}");
-                            return;
-                        }
-                    };
-                    // Create UMessage
-                    invoke_block_callback(
-                        &resp_callback,
-                        UMessage {
-                            attributes: Some(u_attribute).into(),
-                            payload: Some(sample.payload().into()),
-                            ..Default::default()
-                        },
-                    );
-                }
-                Err(e) => {
-                    warn!("Unable to parse Zenoh reply: {e:?}");
-                }
-            }
-        };
-
-        // Send query
-        let getbuilder = self
-            .session
-            .get(zenoh_key)
-            .payload(payload)
-            .attachment(attachment)
-            .target(QueryTarget::BestMatching)
-            .timeout(Duration::from_millis(u64::from(
-                attributes.ttl.unwrap_or(1000),
-            )))
-            .callback(zenoh_callback);
-        getbuilder.await.map_err(|e| {
-            let msg = format!("Unable to send get with Zenoh: {e:?}");
-            error!("{msg}");
-            UStatus::fail_with_code(UCode::INTERNAL, msg)
-        })?;
-
-        Ok(())
-    }
-
-    async fn send_response(&self, payload: Bytes, attributes: UAttributes) -> Result<(), UStatus> {
-        // Transform UAttributes to user attachment in Zenoh
-        let Ok(attachment) = UPTransportZenoh::uattributes_to_attachment(&attributes) else {
-            let msg = "Unable to transform UAttributes to attachment".to_string();
-            error!("{msg}");
-            return Err(UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg));
-        };
-
-        // Find out the corresponding query from HashMap
-        let reqid = attributes.reqid.to_string();
-        let query = self
-            .query_map
-            .lock()
-            .unwrap()
-            .remove(&reqid)
-            .ok_or_else(|| {
-                let msg = "query doesn't exist".to_string();
-                error!("{msg}");
-                UStatus::fail_with_code(UCode::INTERNAL, msg)
-            })?
-            .clone();
-
-        // Send back the query
-        query
-            .reply(query.key_expr().clone(), payload)
             .attachment(attachment)
             .await
-            .map_err(|e| {
-                let msg = format!("Unable to reply with Zenoh: {e:?}");
-                error!("{msg}");
-                UStatus::fail_with_code(UCode::INTERNAL, msg)
-            })?;
+            .map_err(|_| UStatus::fail_with_code(UCode::INTERNAL, "Unable to send with Zenoh"))?;
```

Comment:
I'd prefer to get the error message from Zenoh. I've been in circumstances where I needed on my fork for this repo to add _even more logging_ to understand what I goofed up.